codegen/settingsgen.py

```diff
@@ -340,7 +340,10 @@ def _populate_classes(parent_dir):
             doc = fix_settings_doc(cls.__doc__)
             # Custom doc for child object type
             if cls.fluent_name == "child-object-type":
-                doc = f"'child_object_type' of {file_name[: file_name.find('_child')]}."
+                parent_name = Path(file_name).stem[
+                    0 : Path(file_name).stem.find("_child")
+                ]
+                doc = f"'child_object_type' of {parent_name}."
```

Comment:
Example, before:
`'child_object_type' of E:\ansysdev\pyfluent\src\ansys\fluent\core\solver\settings_232\aeromechanics.`
With these changes:
`'child_object_type' of aeromechanics.`